src/core/lightning/index.ts

```diff
@@ -566,37 +568,25 @@ export const LightningMixin = (superclass) =>
           // even if the payment is still ongoing from lnd.
           // to clean pending payments, another cron-job loop will run in the background.
 
+          let result: PayInvoiceResult | LightningServiceError
           try {
-            // Fixme: seems to be leaking if it timeout.
-            if (route) {
-              paymentPromise = payViaRoutes({ lnd, routes: [route], id })
-            } else {
-              // incoming_peer?
-              // max_paths for MPP
-              // max_timeout_height ??
-              paymentPromise = payViaPaymentDetails({
-                lnd,
-                id,
-                cltv_delta,
-                destination,
-                features,
-                max_fee,
-                mtokens,
-                payment,
-                routes: routeHint,
-              })
-            }
-
-            await Promise.race([paymentPromise, timeout(TIMEOUT_PAYMENT, "Timeout")])
-            // FIXME
-            // return this.payDetail({
-            //     pubkey: details.destination,
-            //     hash: details.id,
-            //     amount: details.tokens,
-            //     routes: details.routes
-            // })
+            const decodedInvoice = decodeInvoice(params.invoice as EncodedPaymentRequest)
+            if (decodedInvoice instanceof Error) throw decodedInvoice
+
+            const lndService = LndService()
+            if (lndService instanceof Error) throw lndService
+
+            result = await lndService.payInvoice({
+              paymentHash: decodedInvoice.paymentHash,
+              rawRoute: route,
+              lndAuthForRoute: lnd,
```

Comment:
lnd should not be part of the interface of `ILightningService`. Its an interface precisely to hide the implementation details. That we are using lnd is an implementation detail. Maybe its just naming - or perhaps there is a boundary issue. Not sure.